isce2_topsapp/iono_proc.py

```diff
@@ -0,0 +1,383 @@
+                   saveFlag: bool = False) -> NDArray:
+    '''
+    This routine translates mask raster from geographical to radar(image) coordinate space
+
+    maskFilename : str
+            path to georeferenced mask raster (must have vrt extension)
+    latFilename : str
+            path to radar lat.rdr file (bursts: geom_reference/IW{1,2,3}/lat_burst.rdr
+                                        merged: merged/lat.rdr.full.vrt)
+    lonFilename : str
+            path to radar lon.rdr file (bursts: geom_reference/IW{1,2,3}/lon_burst.rdr
+                                        merged: merged/lon.rdr.full.vrt)
+    outputFilename : str
+            path to save mask output file
+    saveFlag : bool
+            flag to locally save mask in radar coordinates
+    '''
+
+    # Open mask file
+    mask_ds = gdal.Open(maskFilename + '.vrt')
+    # Open lon and lat file
+    lon_ds = gdal.Open(lonFilename + '.vrt')
+    lat_ds = gdal.Open(latFilename + '.vrt')
+
+    # Get lon and lat arrays
+    lons = lon_ds.ReadAsArray()
+    lats = lat_ds.ReadAsArray()
+
+    # Translate mask geo corrdinate to radar
+    lineIndex = np.int32((lats - mask_ds.GetGeoTransform()[3]) / mask_ds.GetGeoTransform()[5] + 0.5)
+    sampleIndex = np.int32((lons - mask_ds.GetGeoTransform()[0]) / mask_ds.GetGeoTransform()[1] + 0.5)
+    inboundIndex = np.logical_and(
+                np.logical_and(lineIndex >= 0, lineIndex <= mask_ds.RasterYSize-1),
+                np.logical_and(sampleIndex >= 0, sampleIndex <= mask_ds.RasterXSize-1)
+                )
+    # Convert
+    mask_radarcoord = np.empty(lats.shape)
+    mask_radarcoord[inboundIndex] = mask_ds.ReadAsArray()[lineIndex[inboundIndex],
+                                                            sampleIndex[inboundIndex]]
+    # close gdal instances
+    mask_ds = None
+    lon_ds = None
+    lat_ds = None
+
+    # Save
+    if saveFlag:
+        mask_radarcoord.astype(np.int8).tofile(outputFilename)
+        image = isceobj.createImage()
+        image.setDataType('BYTE')
+        image.setFilename(outputFilename)
+        image.extraFilename = outputFilename + '.vrt'
+        image.setWidth(mask_radarcoord.shape[1])
+        image.setLength(mask_radarcoord.shape[0])
+        image.renderHdr()
+
+    return mask_radarcoord
+
+
+def mask_interferogram(ifgFilename: Union[str, Path],
+                       maskArray: NDArray,
+                       outFilename: Union[str, Path] = None) -> None:
+    
+    '''
+    This routine uses mask np.array to mask wrapped interferogram
+    ifgFilename : str
+            path to georeferenced mask raster (must have vrt extension)
+    maskArray : np.array
+            mask 2d array (1: mask values, 0; no-mask values))
+    outFilename : str
+            path where to save masked interferogram (default: None = overwrite existing)
+    '''
+
+    # Read interferogram
+    int_ds = gdal.Open(ifgFilename + '.vrt')
+    int_array = int_ds.ReadAsArray()
```

Comment:
I would make sure the dimensions are the same as well and if not, raise ValueError.

Reply:
solved